### ISSUE TITLE
Create Facet template that indicates when user is in a voice mode other than normal/whisper

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#2169]
### Describe the Underlying Issue

When user is muted, in shout or broadcast, there is not way for them to tell unless they open dash or facet anchors.

### Expected Behavior

Build a facet that specifically indicates when the user's voice mode is anything other than normal/whisper. That is it will show up when the user is muted, in shout or broadcast.

Otherwise the facet visual needs to be completely invisible (make sure it's actually disabled so there's no collisions).

**Comment by RyuviTheViali** [comment]
I've created a basic visual for this issue - it's a small visual indicator that only shows when your voice mode is not whisper or normal. The visual also animates when you talk, and when you swap voice modes for a brief period.

https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/15852289/aacf869f-4777-427d-bd38-33c6ff86404c

There is one related issue that I've noticed, though - disabling the collider (or the whole facet) when on the avatar view facet anchors doesn't stop the userspace laser from interacting with it, though this may deserve it's own issue so as to not go off topic.

**Comment by shiftyscales** [comment]
I don't think it is off-topic in this case, @RyuviTheViali - it was a part of the initial request from @Frooxius to do so- if for some reason it is not working as expected, he'd probably need to look into it and figure out why.

I have noticed a similar issue with context menus as of late. If the user has their lasers active (trigger soft-pulled, or fully active), it will stick to the context menu canvas after the context menu is closed.

I think it may be an issue related to #2153.

**Comment by RyuviTheViali** [comment]
Just an update here, the facet is basically ready, however the above issue where userspace lasers still show on the facet even when its collider is disabled is still present.

**Comment by Frooxius** [comment]
You don't disable the collider - you need to fully disable all the visuals (and you shouldn't be disabling the collider).

The laser passthrough isn't determined by the collider, but rather if the laser hits anything visible on the canvas.

**Comment by shiftyscales** [comment]
As far as I can tell based on @RyuviTheViali's reply both were tried- perhaps they can elaborate.

> disabling the collider (or the whole facet)

**Comment by RyuviTheViali** [comment]
I'll give a shot at disabling all visuals and report back.

**Comment by RyuviTheViali** [comment]
Okay, I've changed the facet so it completely hides the canvas when not needed, and even when disabled like this, my userspace laser still gets triggered. I also attempted disabling the whole facet, with the same outcome.

**Comment by TisFoolish** [comment]
Is this and the context menu issue mentioned the same as lasers getting stuck on keyboards briefly after they've been closed?

**Comment by Frooxius** [comment]
@RyuviTheViali Oh one more thing. Is the "LaserPassThrough" checked on the canvas for the facet?

**Comment by RyuviTheViali** [comment]
@Frooxius Just tested with laser passthrough enabled and my userspace laser still appears.

**Comment by shiftyscales** [comment]
I suspect it might be, but I imagine it'd need to be looked into to know for sure. I'd still recommend opening a new issue about that as it currently doesn't have an issue to track it specifically.

**Comment by Frooxius** [comment]
@RyuviTheViali Can you have a copy of it somewhere so I can have a look please?

**Comment by RyuviTheViali** [comment]
@Frooxius I have sent you a copy of the facet in Resonite messages.

**Comment by Frooxius** [comment]
@RyuviTheViali Could you actually place it in a team folder somewhere please? I have lost the item.

Ideally create one in the root like "Replication Items" or something like that. That way I get it any any time.

**Comment by RyuviTheViali** [comment]
@Frooxius I have placed the facet in: `G-Resonite-Team/Projects/Facets/Voice Mode Indicator`.
